### PR TITLE
Optional LDAP testsupport

### PR DESCRIPTION
**Pull request by mcculls**
Based on the modern LDAP test code from clm-land

### CODE REVIEW

**Comment by cstamas** [comment]
+1

File: testsupport/src/main/java/org/sonatype/sisu/litmus/testsupport/ldap/LdapServer.java [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+
+    Partition acmeBrickPartition = partitionFactory.createPartition(directoryService.getSchemaManager(),
+        directoryService.getDnFactory(), "acme_brick", "dc=acme brick,dc=com", 500, new File(directoryService
+            .getInstanceLayout().getPartitionsDirectory(), "acme_brick"));
+    acmeBrickPartition.setSchemaManager(directoryService.getSchemaManager());
+    partitionFactory.addIndex(acmeBrickPartition, SchemaConstants.OBJECT_CLASS_AT, 100);
+    directoryService.addPartition(acmeBrickPartition);
+  }
+
+  private static int getRandomPort() throws IOException {
```
Comment (cstamas) [inline testsupport/src/main/java/org/sonatype/sisu/litmus/testsupport/ldap/LdapServer.java]: Minor: why not PortRegistry or try-with-resource at least?

Comment (mcculls) [inline testsupport/src/main/java/org/sonatype/sisu/litmus/testsupport/ldap/LdapServer.java]: Should be straightforward to switch it to use `PortRegistry` (this code is unchanged from clm-land)

Also note that the goodies project is still set to Java6 language level, I'll open a separate PR to bump it to Java7 unless there's a good reason to keep it back at Java6.

Comment (mcculls) [inline testsupport/src/main/java/org/sonatype/sisu/litmus/testsupport/ldap/LdapServer.java]: 9ae4fdc

Comment (mcculls) [inline testsupport/src/main/java/org/sonatype/sisu/litmus/testsupport/ldap/LdapServer.java]: Opened https://github.com/sonatype/sisu-goodies/pull/36 to move goodies to Java7

**Pull request merged**